cli/deployment/order.go

```diff
@@ -0,0 +1,137 @@
+package deployment
+
+import (
+	"errors"
+	"fmt"
+	"log"
+)
+// Order function receives map of configs with Config.FullName() string as a key,
+// find dependencies between them, and order them topologically using directed graph.
+// Returns array of ordered config's objects and error if configs have circle dependencies
+func Order(configs map[string]Config) ([]Config, error) {
+	size := len(configs)
+
+	nodes := make([]string, 0)
+	// we don't know number or dependencies, so initial size is 0
+	edges := make([]edge, 0)
+	for _, config := range configs {
+		nodes  = append(nodes, config.FullName())
+		deps := config.findAllDependencies(configs)
+		for _, dep := range deps {
+			edges = append(edges, edge {
+					from:dep.FullName(),
+					to:config.FullName(),
+			})
+		}
+	}
+
+	graph, err := newDirectedGraph(nodes, edges)
+	if err != nil {
+		log.Printf("error creating dependecy grahp: %v", err)
+		return nil, err
+	}
+
+	sorted, err := graph.topologicalSort()
+	if err != nil {
+		log.Printf("error ordering configs: %v", err)
+		return nil, err
+	}
+	res := make([]Config, size)
+	for i, name := range sorted {
+		res[i] = configs[name]
+	}
+	return res, nil
+}
+
+
+// directedGraph struct used to build graph of cross-config depencencies
+// and do topological sort to define ordering of deployment creation
```

Comment:
Note I'm not commenting on the grammatical structure of these comments because they are attached to non-exported types and functions, but you might like to address these as well to assist with readability for folks developing the package itself.